### DISCUSSION TITLE
UnionDoc doesn't support link

### DISCUSSION THREAD

**Discussion by hgalytoby** [BeanieODM/beanie#384] [Question] [answered]
I want to search all the data with UnionDoc.

But some fields of the data are Link.

It looks like UnionDoc doesn't have a fetch_all_links method.

Wondering why there is no fetch_all_links method?

`beanie.exceptions.NotSupported: UnionDoc doesn't support link fetching`

**Comment by Vitalium** [comment, marked as answer]
Hello! Yes, UnionDoc has some limitations. Also, the UnionDoc itself cannot be a Link.

I currently working on [improved inheritance](https://github.com/roman-right/beanie/pull/380). UnionDoc is still supported for backward compatibility.

Here is [almost ready solution](https://github.com/Vitalium/beanie) (Beanie fork). Hope @roman-right will merge it soon.

Documentation is also [here](https://github.com/Vitalium/beanie/blob/main/docs/tutorial/inheritance.md).

**Reply by hgalytoby** [reply]
> Hello! Yes, UnionDoc has some limitations. Also, the UnionDoc itself cannot be a Link.
> 
> I currently working on [improved inheritance](https://github.com/roman-right/beanie/pull/380). UnionDoc is still supported for backward compatibility.
> 
> Here is [almost ready solution](https://github.com/Vitalium/beanie) (Beanie fork). Hope @roman-right will merge it soon.
> 
> Documentation is also [here](https://github.com/Vitalium/beanie/blob/main/docs/tutorial/inheritance.md).

Thank you for the fork, it worked perfectly for me after using it.
